### ISSUE TITLE
When selecting video new instance of VLC opens and immediately closes

### ISSUE THREAD

**Issue by matecrn** [trizen/youtube-viewer#244]
I am currently running Debian Testing (Buster) and I have had some issues with youtube-viewer. Previously, I was also getting the same issues on Debian 9 Stretch too (I only upgraded this installation a few days ago).

When I select a video to play it displays relevant video info in terminal and then attempts to open a new instance of VLC and immediately closes it afterwards. This is the error that is displayed:

```
VLC media player 3.0.6 Vetinari (revision 3.0.6-0-g5803e85f73)
QObject::~QObject: Timers cannot be stopped from another thread
```

After this error is displayed then it loops back to the search prompt.

I can still play videos perfectly fine if I copy the URL and launch it out of VLC though.

Also it is worth noting that I was never able to install all of the recommended Perl modules too, for some reason the Perl module Term::ReadLine::Gnu::XS never installed properly so maybe that has something to do with this.

Whenever I attempt to install Term::ReadLine::Gnu::XS it shows the following:

```
cpan Term::ReadLine::Gnu::XS
Reading '/home/mate/.local/share/.cpan/Metadata'
  Database was generated on Fri, 01 Mar 2019 18:41:02 GMT
CPAN: LWP::UserAgent loaded ok (v6.36)
Fetching with LWP:
http://www.cpan.org/authors/01mailrc.txt.gz
CPAN: YAML loaded ok (v1.27)
Reading '/home/mate/.local/share/.cpan/sources/authors/01mailrc.txt.gz'
CPAN: Compress::Zlib loaded ok (v2.074)
............................................................................DONE
Fetching with LWP:
http://www.cpan.org/modules/02packages.details.txt.gz
Reading '/home/mate/.local/share/.cpan/sources/modules/02packages.details.txt.gz'
  Database was generated on Sat, 02 Mar 2019 21:54:44 GMT
.............
  New CPAN.pm version (v2.22) available.
  [Currently running version is v2.20]
  You might want to try
    install CPAN
    reload cpan
  to both upgrade CPAN.pm and run the new version without leaving
  the current session.


...............................................................DONE
Fetching with LWP:
http://www.cpan.org/modules/03modlist.data.gz
Reading '/home/mate/.local/share/.cpan/sources/modules/03modlist.data.gz'
DONE
Writing /home/mate/.local/share/.cpan/Metadata
Running install for module 'Term::ReadLine::Gnu::XS'
CPAN: Digest::SHA loaded ok (v6.01)
Checksum for /home/mate/.local/share/.cpan/sources/authors/id/H/HA/HAYASHI/Term-ReadLine-Gnu-1.36.tar.gz ok
CPAN: CPAN::Meta::Requirements loaded ok (v2.140)
CPAN: Parse::CPAN::Meta loaded ok (v2.150010)
CPAN: CPAN::Meta loaded ok (v2.150010)
CPAN: Module::CoreList loaded ok (v5.20181129_28)
Configuring H/HA/HAYASHI/Term-ReadLine-Gnu-1.36.tar.gz with Makefile.PL
Could not find neither libtermcap, libtinfo, libncurses, or libcurses.
Warning: No success on command[/usr/bin/perl Makefile.PL INSTALLDIRS=site]
  HAYASHI/Term-ReadLine-Gnu-1.36.tar.gz
  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- NOT OK
```

**Comment by trizen** [comment]
Try to remove the `--quiet` argument in the configuration file to see what errors you get.

Also make sure that `youtube-dl` is installed. Otherwise, some videos cannot be played.

Alternatively, if `vlc [youtube-url]` works, then you can consider adding the following player-entry in the configuration file of youtube-viewer:

```perl
  video_players => {
    # ...
    vlc_native => {
                      arg => "*URL*",
                      cmd => "vlc",
                      fs => "--fullscreen",
                      novideo => "--intf dummy --novideo",
                      srt => "--sub-file *SUB*",
                  },
},
```

...then set the value of `video_player_selected` to `"vlc_native"`:

```perl
  video_player_selected  => "vlc_native",
```

Regarding the `Term::ReadLine::Gnu::XS` module:

> Could not find neither libtermcap, libtinfo, libncurses, or libcurses.

You need to have the `readline` library installed (which may be named `libreadline` or `libreadline5` or `libreadline6`) and the `tinfo` library (which may be named `libtinfo` or `libtinfo5`)  in order to build this module.

If the module is available in the repo, better to install it from there. It may be named `libterm-readline-gnu-perl`.

Let me know if this solves the issue. Thanks.

**Comment by matecrn** [comment]
Removing --quiet did nothing. 

I have youtube-dl installed.

I tried using vlc_native and that didn't work either. I did however get a different error with it:

```
Your input can't be opened:
VLC is unable to open the MRL '[youtube video url]'. Check the log for details.
```

I installed `libterm-readline-gnu-perl` and when I attempted to reinstall youtube-viewer there were no issues this time. 

VLC still doesn't seem to work, however when I change the value of `video_player_selected` to `"mpv"` it works just fine. I also came across [this thread](https://github.com/trizen/youtube-viewer/issues/211) where another user had a similar issue with VLC.

I would prefer VLC, but if only mpv works it's better than not working. Is there anything else that can be done?
